modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java

```diff
@@ -1791,6 +1791,31 @@ public void testAddObjectEntryWithRelationshipObjectField()
 				HashMapBuilder.<String, Object>put(
 					_objectRelationshipFieldName, objectEntry.getId()
 				).build()));
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			companyId, _objectDefinition1.getClassName(),
+			ResourceConstants.SCOPE_INDIVIDUAL,
+			String.valueOf(objectEntry.getId()), role.getRoleId(),
+			new String[] {ActionKeys.VIEW});
+
+		assertEquals(
+			_defaultObjectEntryManager.addObjectEntry(
+				_simpleDTOConverterContext, _objectDefinition2,
+				new ObjectEntry() {
+					{
+						properties = HashMapBuilder.<String, Object>put(
+							_objectRelationshipFieldName, objectEntry.getId()
+						).build();
+					}
+				},
+				ObjectDefinitionConstants.SCOPE_COMPANY),
+			new ObjectEntry() {
+				{
+					properties = HashMapBuilder.<String, Object>put(
+						_objectRelationshipFieldName, objectEntry.getId()
+					).build();
+				}
+			});
```

Comment:
@vngoliveira can you move this change to the `LPD-44832 Add integration test` commit?